packages/lesswrong/server/resolvers/postResolvers.ts

```diff
@@ -350,9 +367,38 @@ augmentFieldsDict(Posts, {
       },
     },
   },
+
+  glossary: {
+    resolveAs: {
+      type: GraphQLJSON,
+      resolver: async (post: DbPost, args: void, context: ResolverContext) => {
+        // Forum-gating/beta-gating is done here, rather than just client side,
+        // so that users don't have to download the glossary if it isn't going
+        // to be displayed.
+        if (!userCanViewJargonTerms(context.currentUser)) {
+          return;
+        }
+        const jargonTerms = await context.JargonTerms.find({postId: post._id, rejected: false, deleted: false}).fetch();
```

Comment:
I'd want to sanity check that we aren't fetching the glossary for the same post multipe times in one request (since if we do that, for a reason that's difficult to work around, we might want to create a custom loader for this).